src/modules/visitor-space/components/CreatorFilterForm/CreatorFilterForm.tsx

```diff
@@ -27,28 +34,56 @@ const CreatorFilterForm: FC<CreatorFilterFormProps> = ({ children, className })
 	const { tHtml, tText } = useTranslation();
 
 	// State
-
 	const [query] = useQueryParams(CREATOR_FILTER_FORM_QUERY_PARAM_CONFIG);
+	const [currentSearchFilters] = useQueryParams(VISITOR_SPACE_QUERY_PARAM_CONFIG);
 	const [search, setSearch] = useState<string>('');
 	const [selection, setSelection] = useState<string[]>(() => compact(query.creator || []));
+	const [isSearchEnabled, setIsSearchEnabled] = useState<boolean>(false);
 
 	const { setValue, reset, handleSubmit } = useForm<CreatorFilterFormState>({
 		resolver: yupResolver(CREATOR_FILTER_FORM_SCHEMA()),
 		defaultValues,
 	});
 
-	const buckets = (
-		useSelector(selectIeObjectsFilterOptions)?.schema_creator?.buckets || []
-	).filter((bucket) => bucket.key.toLowerCase().includes(search.toLowerCase()));
+	const filters: IeObjectsSearchFilter[] = [
+		...mapFiltersToElastic(currentSearchFilters),
+		...mapRefineFilterToElastic([
+			{
+				field: IeObjectsSearchFilterField.CREATOR,
+				value: search,
+			},
+		]),
+	];
+
+	const initialAggregates = useSelector(selectIeObjectsFilterOptions);
+	const { data: aggregates } = useGetIeObjectsAggregatesByField(
+		VisitorSpaceFilterId.Creator,
+		filters,
+		isSearchEnabled
+	);
 
-	// Effects
+	const buckets = useMemo(() => {
+		// TODO: add sort
```

Comment:
Do we need to implement this todo/task now or later?

Reply:
Those are sorted by most results to fewest results. So if we sort them alphabetically it might be more confusing, since then you get the 40 maintainers with most results sorted alphabetically. While you would expect the first 40 maintainers alphabetically.

I'll revisit the sorting when i implement searching for checkbox options the modal